src/test/regress/sql/pg16.sql

```diff
@@ -650,10 +624,92 @@ RESET citus.enable_ddl_propagation;
 -- by default, admin option is false, inherit is true, set is true
 GRANT role3 TO role4;
 GRANT role3 TO role5 WITH ADMIN TRUE, INHERIT FALSE, SET FALSE;
-SELECT roleid::regrole::text AS role, member::regrole::text, admin_option, inherit_option, set_option FROM pg_auth_members WHERE roleid::regrole::text = 'role3' ORDER BY 1, 2;
+SELECT roleid::regrole::text AS role, member::regrole::text, admin_option, inherit_option, set_option FROM pg_auth_members
+WHERE roleid::regrole::text = 'role3' ORDER BY 1, 2;
 
 DROP ROLE role3, role4, role5;
 
+-- Test that everything works fine for roles that are propagated
+CREATE ROLE role6;
+CREATE ROLE role7;
+CREATE ROLE role8;
+CREATE ROLE role9;
+CREATE ROLE role10;
+CREATE ROLE role11;
+CREATE ROLE role12;
+CREATE ROLE role13;
+CREATE ROLE role14;
+CREATE ROLE role15;
+CREATE ROLE role16;
+CREATE ROLE role17;
+CREATE ROLE role18 NOINHERIT;
+CREATE ROLE role19;
+CREATE ROLE role20;
+
+-- Grant role with admin and inherit options set to true
+GRANT role6 TO role7 WITH ADMIN OPTION, INHERIT TRUE;
+-- GRANT with INHERIT and SET Options
+-- note that set is true by default so we don't include it in the propagation
+GRANT role7 TO role8 WITH INHERIT TRUE, SET TRUE;
+-- Grant role with admin option set to true and inherit option set to false
+GRANT role9 TO role10 WITH ADMIN OPTION, INHERIT FALSE;
+-- Grant role with admin option set to true, and inherit/set options set to false
+GRANT role11 TO role12 WITH INHERIT FALSE, ADMIN TRUE, SET FALSE;
+-- Grant role with inherit set to false
+GRANT role13 TO role14 WITH INHERIT FALSE;
+-- Grant role with set option set to false
+GRANT role15 TO role16 WITH SET FALSE;
+-- Handles with default inherit false
+-- we created role18 with noinherit option above
+GRANT role17 TO role18;
+-- Run GRANT/REVOKE commands on worker nodes
+\c - - - :worker_1_port
+-- Run GRANT command on worker node
+GRANT role19 TO role20;
+-- Run REVOKE command on worker node
+REVOKE role19 FROM role20;
+\c - - - :master_port
+
+SELECT roleid::regrole::text AS role, member::regrole::text, admin_option, inherit_option, set_option
+FROM pg_auth_members
+WHERE roleid::regrole::text LIKE 'role%'
+ORDER BY 1, 2;
+\c - - - :worker_1_port
+SELECT roleid::regrole::text AS role, member::regrole::text, admin_option, inherit_option, set_option
+FROM pg_auth_members
+WHERE roleid::regrole::text LIKE 'role%'
+ORDER BY 1, 2;
+
+\c - - - :master_port
+DROP ROLE role6, role7, role8, role9, role10, role11, role12,
+          role13, role14, role15, role16, role17, role18;
```

Comment:
Should we add role19 and role20 to this list to unbreak flaky tests?

```suggestion
DROP ROLE role6, role7, role8, role9, role10, role11, role12,
          role13, role14, role15, role16, role17, role18, role19, role20;
```